src/c++/perf_analyzer/genai-perf/genai_perf/parser.py

```diff
@@ -119,19 +124,47 @@ def _check_conditional_args(
     return args
 
 
-def _update_load_manager_args(args: argparse.Namespace) -> argparse.Namespace:
+def _check_load_manager_args(args: argparse.Namespace) -> argparse.Namespace:
     """
-    Update genai-perf load manager attributes to PA format
+    Check inference load args
     """
-    for attr_key in ["concurrency", "request_rate"]:
-        attr_val = getattr(args, attr_key)
-        if attr_val is not None:
-            setattr(args, f"{attr_key}_range", f"{attr_val}")
-            delattr(args, attr_key)
-            return args
-
     # If no concurrency or request rate is set, default to 1
-    setattr(args, "concurrency_range", "1")
+    if not args.concurrency and not args.request_rate:
+        args.concurrency = 1
+    return args
+
+
+def _set_artifact_paths(args: argparse.Namespace) -> argparse.Namespace:
+    """
+    Set paths for all the artifacts.
+    """
+    if args.artifact_dir == Path(DEFAULT_ARTIFACT_DIR):
+        name = [f"{args.model}"]
+        if args.service_kind == "openai":
+            name += [f"{args.service_kind}-{args.endpoint_type}"]
+        elif args.service_kind == "triton":
```

Comment:
I don't know if you'd be interested in this, but you could update the enum to the below:
```from enum import Enum, auto

class OutputFormat(Enum):
    OPENAI_CHAT_COMPLETIONS = auto()
    OPENAI_COMPLETIONS = auto()
    TENSORRTLLM = auto()
    VLLM = auto()

    def to_lowercase(self):
        return self.name.lower()
```

You would be able to get rid of these branches. You could even get rid of this entire branching tree. That depends on how concerned you are with a specific naming scheme and validation (which should be done elsewhere already).

Reply:
what?! thats awesome

Reply:
@dyastremsky I think that's neat! Updated to your suggestion.